### PR TITLE
fix torchvision installation in tpu ci test setup script

### PR DESCRIPTION
**Pull request by lsy323**
Currently in TPU CI, we are installing torch nightly whl before running tests, however, in TPU CI, torch_xla is built against PyTorch HEAD. This would cause some compatibility issues. (e.g. Symbol mismatching in .so)

- torchaudio is not needed for CI, remove torchaudio installation in the setup script as well.

-  Instead of installing nightly torchvision, use the torchvision version pinned in PyTorch src folder.

### CODE REVIEW

**Comment by lsy323** [comment]
Cannot find a way to read the `pytorch/.github/ci_commit_pins` in `test/tpu/xla_test_job.yaml`.

When doing `pip install "git+https://github.com/pytorch/vision.git@$TORCHVISION_COMMIT"`, `$TORCHVISION_COMMIT` will be an empty str.

To rule out the fact that we cannot change env var in the `.yaml` config files, I put the `pip` command in a shell script but looks like the path `src/pytorch/.github/ci_commit_pins` is not valid.

File: infra/ansible/e2e_tests.Dockerfile [hunk 1]
```diff
@@ -23,6 +23,9 @@ RUN ansible-playbook -vvv playbook.yaml -e "stage=release" -e "${ansible_vars}"
 # Copy test sources.
 RUN mkdir -p /src/pytorch/xla
 COPY --from=build /src/pytorch/xla/test /src/pytorch/xla/test
+# Copy ci_commit_pins from upstream
+RUN mkdir -p /src/pytorch/.github
+COPY --from=build /src/pytorch/.github/ci_commit_pins /src/pytorch/xla/test/ci_commit_pins
```
Comment (will-cromar) [inline infra/ansible/e2e_tests.Dockerfile:28]: nit: can you run the torchvision install in the Dockerfile?

**Review by will-cromar — COMMENTED** [review]
Can you also mirror the fix to the GHA CI if it works? https://github.com/pytorch/xla/blob/master/.github/workflows/tpu_ci.yml

The commit pin will be more readily available.

File: infra/ansible/e2e_tests.Dockerfile (hunk 1 above)
Comment (lsy323) [reply infra/ansible/e2e_tests.Dockerfile]: Tried to install in Dockerfile, but looks like installing torchvision in this way requires torch to be already installed. So we have to install after torch is compiled.
```
Step #2 - "build_pytorch-xla-test_docker_image": Step 19/26 : RUN pip install --no-use-pep517 "git+https://github.com/pytorch/vision.git@$(cat /src/pytorch/.github/ci_commit_pins/vision.txt)"
Step #2 - "build_pytorch-xla-test_docker_image":  ---> Running in 5513d8dbc80a
Step #2 - "build_pytorch-xla-test_docker_image": Collecting git+https://github.com/pytorch/vision.git@2c127da8b5e2e8f44b50994c6cb931bcca267cfe
Step #2 - "build_pytorch-xla-test_docker_image":   Cloning https://github.com/pytorch/vision.git (to revision 2c127da8b5e2e8f44b50994c6cb931bcca267cfe) to /tmp/pip-req-build-th1jpkpj
Step #2 - "build_pytorch-xla-test_docker_image": �[91m  Running command git clone --filter=blob:none --quiet https://github.com/pytorch/vision.git /tmp/pip-req-build-th1jpkpj
Step #2 - "build_pytorch-xla-test_docker_image": �[0m�[91m  Running command git rev-parse -q --verify 'sha^2c127da8b5e2e8f44b50994c6cb931bcca267cfe'
Step #2 - "build_pytorch-xla-test_docker_image": �[0m�[91m  Running command git fetch -q https://github.com/pytorch/vision.git 2c127da8b5e2e8f44b50994c6cb931bcca267cfe
Step #2 - "build_pytorch-xla-test_docker_image": �[0m�[91m  Running command git checkout -q 2c127da8b5e2e8f44b50994c6cb931bcca267cfe
Step #2 - "build_pytorch-xla-test_docker_image": �[0m  Resolved https://github.com/pytorch/vision.git to commit 2c127da8b5e2e8f44b50994c6cb931bcca267cfe
Step #2 - "build_pytorch-xla-test_docker_image":   Preparing metadata (setup.py): started
Step #2 - "build_pytorch-xla-test_docker_image":   Preparing metadata (setup.py): finished with status 'error'
Step #2 - "build_pytorch-xla-test_docker_image": �[91m  error: subprocess-exited-with-error
Step #2 - "build_pytorch-xla-test_docker_image":   
Step #2 - "build_pytorch-xla-test_docker_image":   × python setup.py egg_info did not run successfully.
Step #2 - "build_pytorch-xla-test_docker_image":   │ exit code: 1
Step #2 - "build_pytorch-xla-test_docker_image":   ╰─> [6 lines of output]
Step #2 - "build_pytorch-xla-test_docker_image":       Traceback (most recent call last):
Step #2 - "build_pytorch-xla-test_docker_image":         File "<string>", line 2, in <module>
Step #2 - "build_pytorch-xla-test_docker_image":         File "<pip-setuptools-caller>", line 34, in <module>
Step #2 - "build_pytorch-xla-test_docker_image":         File "/tmp/pip-req-build-th1jpkpj/setup.py", line 9, in <module>
Step #2 - "build_pytorch-xla-test_docker_image":           import torch
Step #2 - "build_pytorch-xla-test_docker_image":       ModuleNotFoundError: No module named 'torch'
Step #2 - "build_pytorch-xla-test_docker_image":       [end of output]
Step #2 - "build_pytorch-xla-test_docker_image":   
Step #2 - "build_pytorch-xla-test_docker_image":   note: This error originates from a subprocess, and is likely not a problem with pip.
Step #2 - "build_pytorch-xla-test_docker_image": �[0m�[91merror: metadata-generation-failed
Step #2 - "build_pytorch-xla-test_docker_image": 
Step #2 - "build_pytorch-xla-test_docker_image": × Encountered error while generating package metadata.
Step #2 - "build_pytorch-xla-test_docker_image": ╰─> See above for output.
Step #2 - "build_pytorch-xla-test_docker_image": 
Step #2 - "build_pytorch-xla-test_docker_image": note: This is an issue with the package mentioned above, not pip.
Step #2 - "build_pytorch-xla-test_docker_image": hint: See above for details.
Step #2 - "build_pytorch-xla-test_docker_image": �[0m�[91m
Step #2 - "build_pytorch-xla-test_docker_image": [notice] A new release of pip is available: 23.0.1 -> 24.0
Step #2 - "build_pytorch-xla-test_docker_image": [notice] To update, run: pip install --upgrade pip
Step #2 - "build_pytorch-xla-test_docker_image": The command '/bin/sh -c pip install --no-use-pep517 "git+https://github.com/pytorch/vision.git@$(cat /src/pytorch/.github/ci_commit_pins/vision.txt)"' returned a non-zero code: 1
```

**Comment by lsy323** [comment]
> Can you also mirror the fix to the GHA CI if it works? https://github.com/pytorch/xla/blob/master/.github/workflows/tpu_ci.yml
> 
> The commit pin will be more readily available.

I had a few attempts in the previous commits fixing the GHA TPU CI, but failed to do so, let's land the fix for installing torchvision properly for the old TPU CI.

**Review by JackCaoG — APPROVED** [review]

File: infra/ansible/e2e_tests.Dockerfile (hunk 1 above)
Comment (will-cromar) [reply infra/ansible/e2e_tests.Dockerfile]: I guess that's fine. The GCB-based CI is on its way out anyway

File: .github/workflows/tpu_ci.yml [hunk 2]
```diff
@@ -33,7 +31,6 @@ jobs:
           PJRT_DEVICE: TPU
         # Jax is needed for pallas tests.
         run: |
-          pip install --pre torch torchvision --index-url https://download.pytorch.org/whl/nightly/cpu --user
```
Comment (will-cromar) [inline .github/workflows/tpu_ci.yml:36]: Are these changes intentional?

Comment (lsy323) [reply .github/workflows/tpu_ci.yml:36]: Thanks for catching it, I need to undo the change to GHA

Comment (lsy323) [reply .github/workflows/tpu_ci.yml]: Done recover to HEAD

**Comment by lsy323** [comment]
> https://github.com/pytorch/xla/blob/master/.github/workflows/tpu_ci.yml

GHA TPU CI fixed in https://github.com/pytorch/xla/pull/6730
